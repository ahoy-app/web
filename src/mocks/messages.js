import faker from 'faker'

faker.seed(0)

let users = Array.apply(0, Array(6)).map(() => {
  return faker.internet.userName()
})
users.push('-', '-', '-')

const createMoreMessages = () =>
  Array.apply(0, Array(10)).map(() => {
    return {
      id: faker.random.uuid(),
      from: faker.random.arrayElement(users),
      content: faker.lorem.sentences(),
    }
  })

export const fetchOldMessages = (messages = []) => {
  const olderMessages = createMoreMessages()
  messages.unshift(...olderMessages)
  return messages
}

export const receiveNewMessages = (messages = []) => {
  const newerMessages = createMoreMessages()
  messages.push(...newerMessages)
  return messages
}
