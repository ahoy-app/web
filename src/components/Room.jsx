import React from 'react'
import Radium from 'radium'

import Input from '../components/Input'
import Messages from '../components/Messages'

import { receiveNewMessages, fetchOldMessages } from '../mocks/messages'
import { color } from '../style'

const styles = {
  background: color.background.light,

  height: '100%',
  width: '100%',

  display: 'flex',
  flexDirection: 'column',
}

const Room = () => (
  <div style={styles}>
    <Messages
      messages={receiveNewMessages()}
      fetchOldMessages={fetchOldMessages}
    />
    <Input
      onChange={event => {
        console.log('Updated text:' + event.target.value)
      }}
      onSend={() => {
        console.log('Send click')
      }}
    />
  </div>
)

export default Radium(Room)
