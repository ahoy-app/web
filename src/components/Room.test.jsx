import React from 'react'
import { shallow } from 'enzyme'
import * as MockMessages from '../mocks/messages'
import Room from './Room'

describe('Room', () => {
  MockMessages.receiveNewMessages = () => [
    {
      id: '1',
      from: 'a',
      content: 'aaa',
    },
    {
      id: '2',
      from: 'b',
      content: 'bbb',
    },
  ]
  it('should render correctly', () => {
    const component = shallow(<Room />) // eslint-disable-line react/no-children-prop
    expect(component).toMatchSnapshot()
  })
})
