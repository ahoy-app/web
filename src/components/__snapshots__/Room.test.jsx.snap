// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room should render correctly 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "background": "#fff",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
